_posts/2018-09-05-Data-Continuity-Service-for-DynamoDB.md

```diff
@@ -0,0 +1,570 @@
+            "Principal": {
+              "Service": "lambda.amazonaws.com"
+            }
+          }
+        ],
+        "Version": "2012-10-17"
+      },
+      "Path": "/",
+      "Policies": [
+        {
+          "PolicyDocument": {
+            "Statement": [
+              {
+                "Action": [
+                  "logs:CreateLogGroup",
+                  "logs:CreateLogStream",
+                  "logs:DeleteSubscriptionFilter",
+                  "logs:PutLogEvents",
+                  "logs:PutSubscriptionFilter",
+                  "logs:TestMetricFilter"
+                ],
+                "Effect": "Allow",
+                "Resource": "arn:aws:logs:<YOUR_AWS_REGION>:<YOUR_AWS_ACCOUNT_ID>:*"
+              }
+            ]
+          },
+          "PolicyName": "AllowLogs"
+        },
+        {
+          "PolicyDocument": {
+            "Statement": [
+              {
+                "Action": [
+                  "lambda:InvokeFunction"
+                ],
+                "Effect": "Allow",
+                "Resource": "arn:aws:lambda:<YOUR_AWS_REGION>:<YOUR_AWS_ACCOUNT_ID>:function:*"
+              }
+            ]
+          },
+          "PolicyName": "AllowLambaInvoke"
+        },
+        {
+          "PolicyDocument": {
+            "Statement": [
+              {
+                "Action": [
+                  "s3:PutObject"
+                ],
+                "Effect": "Allow",
+                "Resource": "arn:aws:s3:::source-account-dynamodb-bucket/*"
+              }
+            ]
+          },
+          "PolicyName": "AllowS3Access"
+        },
+        {
+          "PolicyDocument": {
+            "Statement": [
+              {
+                "Action": [
+                  "dynamodb:GetRecords",
+                  "dynamodb:GetShardIterator",
+                  "dynamodb:DescribeStream",
+                  "dynamodb:ListStreams"
+                ],
+                "Effect": "Allow",
+                "Resource": "arn:aws:dynamodb:<YOUR_AWS_REGION>:<YOUR_AWS_ACCOUNT_ID>:table/users/stream/*"
+              }
+            ]
+          },
+          "PolicyName": "AllowDynamoDbAccess"
+        }
+      ]
+    },
+    "Type": "AWS::IAM::Role"
+  }
+}
+```
+
+A few things to note. First we are allowing this role to assume on behalf of AWS Lambda by specifying the `sts:AssumeRole` permissions under the "AssumeRolePolicyDocument" key.
+
+Next we are defining some needed policies. We are allowing Lambda to write to CloudWatch logs for logging purposes. We also allowing Lambda to invoke the function or else it wouldn't be able start at all. We are allowing Lambda to write to the source bucket and lastly we allowing Lambda to access and read the DynamoDB Stream which is the hole point of this setup :smiley:
+
+Don't forget to enter the correct AWS Region and your own AWS Account Id.
+
+Once that is in place we can create our Lambda function and assign it our newly created execution role.
```

Comment:
Once that is in place we can create our Lambda function and assign it to our newly created execution role

Reply:
Thanks